appcues/src/main/java/com/appcues/analytics/Analytics.kt

```diff
@@ -0,0 +1,236 @@
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.launch
+import java.util.Date
+import java.util.UUID
+
+internal interface QualificationService {
+
+    // will merge and hit the backend to see if it qualifies, returning QualificationResult
+    suspend fun qualify(intents: List<AnalyticsActivity>): QualificationResult?
+}
+
+internal interface RenderingService {
+
+    // abstraction about who is responsible for tracking experience events
+    interface EventTracker {
+
+        fun trackEvent(name: String, properties: Map<String, Any>?, isInteractive: Boolean, isInternal: Boolean)
+    }
+
+    fun setEventTracker(eventTracker: EventTracker)
+
+    // will attempt to show experiences by order
+    // clearCache is mostly used when we know the qualification is from a screen_view
+    suspend fun show(experiences: List<Experience>, clearCache: Boolean)
+
+    suspend fun reset()
+}
+
+internal interface SessionService {
+
+    // checks session or creates one based on existing information or incoming intent
+    suspend fun checkSession(intent: AnalyticsIntent, onSessionStarted: suspend () -> Unit): Boolean
+
+    // produce session properties
+    suspend fun getSessionProperties(): SessionProperties?
+
+    suspend fun reset()
+}
+
+internal data class SessionProperties(
+    val sessionId: UUID,
+    val userId: String,
+    val groupId: String?,
+    val userSignature: String?,
+    val latestUserProperties: Map<String, Any>,
+    val properties: Map<String, Any>,
+)
+
+internal interface ActivityBuilder {
+
+    suspend fun buildActivity(intent: AnalyticsIntent, sessionProperties: SessionProperties?): AnalyticsActivity?
+}
+
+// new model for Intents available for Analytics
+internal sealed class AnalyticsIntent(open val isInternal: Boolean, open val properties: Map<String, Any>?, val timestamp: Date = Date()) {
+
+    object Anonymous : AnalyticsIntent(false, null)
+
+    data class Identify(
+        val userId: String,
+        override val properties: Map<String, Any>?
+    ) : AnalyticsIntent(false, properties)
+
+    data class UpdateProfile(
+        override val isInternal: Boolean,
+        override val properties: Map<String, Any>?
+    ) : AnalyticsIntent(isInternal, properties)
+
+    data class UpdateGroup(
+        override val isInternal: Boolean,
+        val groupId: String?,
+        override val properties: Map<String, Any>?
+    ) : AnalyticsIntent(isInternal, properties)
+
+    data class Event(
+        override val isInternal: Boolean,
+        val name: String,
+        override val properties: Map<String, Any>?
+    ) : AnalyticsIntent(isInternal, properties)
+
+    data class Screen(
+        override val isInternal: Boolean,
+        val title: String,
+        override val properties: Map<String, Any>?
+    ) : AnalyticsIntent(isInternal, properties)
+}
+
+internal data class AnalyticsActivity(
```

Comment:
this looks similar to ActivityRequest but its supposed to be an improvement over it, at the time when we "map" intents to AnalyticsActivity  we know we always have ONE event instead of a list that we can send on ActivityRequest. this way its easier to map from one to another, and read it through analyticsFlow